docs/code_annotations.rst

```diff
@@ -0,0 +1,166 @@
+Code Annotations
+================
+
+A special directive was created to make use of the mkdocs-material theme's
+`Code Annotations <https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations>`_
+feature.
+
+.. rst:directive:: code-annotations
+
+    This directive shall hold an :duref:`enumerated list <enumerated-lists>` of annotations that can be used in a
+    `code-block` immediately prior to this directive. The given list will not be rendered in the
+    generated output (even if there is no code snippet to annotate). Additionally, all annotations will be
+    excluded when using the "copy to clipboard" button for an annotated `code-block`; the annotated
+    comments' text will still be copied (if not `Hiding the annotated comment`_).
+
+    An error will be shown in the build log if the given list is not recognized as an
+    :duref:`enumerated list <enumerated-lists>`.
+
+    .. note::
+        When using this directive the :python:`"content.code.annotate"` does not need to be
+        specified in the :themeconf:`features` list.
+
+Using annotations
+*****************
+
+Each annotation in the code snippet should be a comment starting with the identifying number
+(in parenthesis) of the corresponding annotation specified in the list of
+:rst:dir:`code-annotations`.
+
+.. note::
+    Remember to specify the language syntax for pygments to highlight the `code-block`. Without
+    this information, pygments may incorrectly identify comments in a `code-block` which is
+    required for the detection of annotations.
+
+.. hint::
+    All annotations are hyperlinked. This means you can right click the annotation's button and
+    open it as a link in a new tab (or copy the link to sharing).
+
+.. warning::
+    Using the same annotation for multiple comments will essentially turn all but the last button
+    into hyperlinks to the last button.
+
+.. rst-example:: Example of code annotations
+
+    .. code-block:: python
+
+        html_theme_options = {
+            "features" : [
+                "content.code.annotate"  # (1)
+            ],
+        }
+
+    .. code-annotations::
+        1. .. admonition:: Obsolete
+               :class: failure
+
+               This has no special affect because the :rst:dir:`code-annotations` directive already
+               enables the feature accordingly.
+
+:duref:`Enumerated lists' <enumerated-lists>` markers in reStructuredText can be uniformed with the
```

Comment:
"specified" instead of "uniformed"?   Since the markers in the code block need to specify the actual number, I'm not sure how helpful this suggestion is, though...

Reply:
Well it turns out that the annotations' identifying number (in the code block) is only related to the corresponding _ordered_ list item (`<ol><li>...`). The number used in the list markers doesn't seem to matter. I think this is why squidfunk changed the buttons to use a generic `+` instead of the list items' numbers.

For example, this should still work:
```rst
.. code-block:: py

    # (1)!
    # (2)!
.. code-annotations::
    3. The first
    #. The second
```

I think I'll play around and proofread a bit more. I want to avoid this confusion.

Reply:
I have embellished this example into a separate section to talk about the ambiguity. The "Annotations buttons with numbers" section now refers this new section as rationality in why the behavior changed upstream.